WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift

```diff
@@ -109,6 +139,27 @@ final class ProductDetailsViewModel {
             self.onError?()
         }
     }
+
+    /// Setup: Results Controller.
+    ///
+    func refreshResultsController() {
+        try? resultsController.performFetch()
+
+        // After refreshing the results controller,
+        // let's look up some product settings it holds.
+        weightUnit = lookupProductSettings(Keys.weightUnit)
+        dimensionUnit = lookupProductSettings(Keys.dimensionUnit)
+    }
+
+    /// Look up Product Settings
+    ///
+    func lookupProductSettings(_ settingID: String) -> String? {
+        for setting in productSettings where setting.settingID == settingID {
```

Comment:
I'm just curious, but do you have any particular reason to prefer this syntax over something like this?
```
return productSettings.filter({$0.settingID == settingID}).first
```
Seriously, it's just curiosity

Reply:
That's so pretty! I forget that `filter` and `map` exist. Going to change this code